RadixWallet/Clients/DappInteractionClient/DappInteractionClient+Live.swift

```diff
@@ -156,3 +161,30 @@ extension DappInteractionClient {
 		)
 	}
 }
+
+extension DappToWalletInteractionProofOfOwnershipRequestItem {
+	func isValid() async -> Bool {
+		@Dependency(\.personasClient) var personasClient
+		@Dependency(\.accountsClient) var accountsClient
+		do {
+			if let identityAddress {
+				// If there is an identityAddress set, verify we can access its corresponding Persona
+				let _ = try await personasClient.getPersona(id: identityAddress)
+			}
+			if let accountAddresses {
+				// For every accountAddress set, verify we can access its corresponding Account
+				let allAccounts = try await accountsClient.getAccountsOnCurrentNetwork()
+				let accounts = allAccounts.filter {
+					accountAddresses.contains($0.address)
+				}
+				if accounts.count != accountAddresses.count {
+					return false
```

Comment:
yes, it will be invalid because one cannot prove ownership about hidden accounts/personas, and if even one entity is missing then we can stop the interaction at this point

@GhenadieVP has confirmed this on last design session as well.

Reply:
yup, the response is all or nothing, no partial response.